src/functions/text.js

```diff
@@ -30,9 +31,11 @@ const optionsWithFont = (group, options) => {
         return options.withTextFontFamily(textFontFamilies[font]);
     } else if (textFontWeights[font]) {
         return options.withTextFontWeight(textFontWeights[font]);
-    } else {
-        return options.withTextFontShape(textFontShapes[font]);
+    } else if (options.fontShape === "textit" && font === "\\emph") {
```

Comment:
I'd suggest changing this to `if (font === "\emph")` and using `options.withTextFontShape` (enforcing text mode) with `"textit"` or `"textup"` according to `options.fontShape`.